pkg/workloads/kubernetes.go

```diff
@@ -62,5 +62,10 @@ func fetchK8sLabels(containerLbls map[string]string) (map[string]string, error)
 		k8sLabels[policy.JoinPath(k8sConst.PodNamespaceMetaLabels, k)] = v
 	}
 	k8sLabels[k8sConst.PodNamespaceLabel] = ns
+
+	if result.Spec.ServiceAccountName != "" {
+		k8sLabels[k8sConst.PolicyLabelServiceAccount] = result.Spec.ServiceAccountName
+	}
```

Comment:
Prevent users from spoofing a service account by defining this label in specs:
```
} else {
    delete k8sLabels[k8sConst.PolicyLabelServiceAccount]
}
```

Reply:
Good idea I think but it doesn't prevent spoofing as the service account name can be freely specified in the pod spec so I'm not sure what security this increases. There is also "default" service account associated if nothing is specified so it will likely be dead code but probably the right thing to be sure. I think we will need an admission controller to make both service account and labels reliable.

Reply:
Maybe. E.g. the Istio docs recommend to use Kubernetes's recommended admission controllers set, which includes `ServiceAccount`. But that admission controller doesn't control the usage of service accounts. And I've not found anything that can restrict the usage of service accounts. AFAICT, if a user has the right to create workloads in a namespace, she can use any service account that exists in that namespace.
Let's leave it as is...

Reply:
The corresponding admission controller does not exist yet but it is fairly trivial to write. It already exists for other privileges such as resources, privileged execution, etc.